dynawo/sources/Modeler/Common/DYNConnector.cpp

```diff
@@ -665,7 +779,10 @@ ConnectorContainer::getY0ConnectorForYConnector() {
             ++it) {
       propertyContinuousVar_t * yType = it->subModel()->getYType();
 
-      if (yType[it->variable()->getIndex()] != EXTERNAL && yType[it->variable()->getIndex()] != OPTIONAL_EXTERNAL) {  // non external variable
+      if (!it->variable()->isExternal() &&
+        !it->subModel()->isFictiveVariableModel() &&
+        yType[it->variable()->getIndex()] != EXTERNAL &&
```

Comment:
isExternal is for Continuous external variables. external discrete variables still have the yType EXTERNAL

Reply:
then testing yType[it->variable()->getIndex()] != EXTERNAL is enough right ? as we want neither continuous nor discrete variables to go in the if at this point

Reply:
we must check that the variable is not external before tetsing the yType because in case of external variable, the index of the variable is the index in the external array, not the y array